crates/daphne/src/pine/vdaf.rs

```diff
@@ -495,9 +517,16 @@ impl<F: FftFriendlyFieldElement> Aggregator<16, 16> for Pine<F> {
                 *v += s1;
             }
 
-            for verifier in verifiers.chunks(self.flp.verifier_len()) {
+            let (v_sq_norm_equal, v) = verifiers.split_at(self.flp_sq_norm_equal.verifier_len());
+            if !self.flp_sq_norm_equal.decide(v_sq_norm_equal)? {
+                return Err(VdafError::Uncategorized(
+                    "squared norm equality proof check failed".into(),
+                ));
+            }
+
+            for verifier in v.chunks(self.flp.verifier_len()) {
                 if !self.flp.decide(verifier)? {
-                    return Err(VdafError::Uncategorized("proof check failed".into()));
+                    return Err(VdafError::Uncategorized("main proof check failed".into()));
```

Comment:
it maybe good to refactor the 'main' proof and give it a more specific name.

Reply:
Unfortunately the functional split between the two FLPs is a little awkward. The "main" circuit is everything but the squared norm equality check. In the future we might try to go for a split for which the two circuits would have more meaningful names: https://github.com/junyechen1996/draft-chen-cfrg-vdaf-pine/issues/95